components/accelerator/nvidia/query/nvml/clock_events.go

```diff
@@ -112,58 +125,201 @@ func GetClockEvents(uuid string, dev device.Device) (ClockEvents, error) {
+		isHWSlowdown: false,
+	},
 
 	// ref. nvmlClocksEventReasonSwPowerCap
-	0x0000000000000004: "Clocks have been optimized to not exceed currently set power limits ('SW Power Cap: Active' in nvidia-smi --query)",
+	reasonSWPowerCap: {
+		description:  "Clocks have been optimized to not exceed currently set power limits ('SW Power Cap: Active' in nvidia-smi --query)",
+		isHWSlowdown: false,
+	},
 
 	// ref. nvmlClocksThrottleReasonHwSlowdown
-	0x0000000000000008: "HW Slowdown is engaged due to high temperature, power brake assertion, or high power draw ('HW Slowdown: Active' in nvidia-smi --query)",
+	reasonHWSlowdown: {
+		description:  "HW Slowdown is engaged due to high temperature, power brake assertion, or high power draw ('HW Slowdown: Active' in nvidia-smi --query)",
+		isHWSlowdown: true,
+	},
 
 	// ref. nvmlClocksEventReasonSyncBoost
-	0x0000000000000010: "GPU is part of a Sync boost group to maximize performance per watt",
+	reasonSyncBoost: {
+		description:  "GPU is part of a Sync boost group to maximize performance per watt",
+		isHWSlowdown: false,
+	},
 
 	// ref. nvmlClocksEventReasonSwThermalSlowdown
-	0x0000000000000020: "SW Thermal Slowdown is active to keep GPU and memory temperatures within operating limits",
+	reasonSwThermalSlowdown: {
+		description:  "SW Thermal Slowdown is active to keep GPU and memory temperatures within operating limits",
+		isHWSlowdown: false,
+	},
 
 	// ref. nvmlClocksThrottleReasonHwThermalSlowdown
-	0x0000000000000040: "HW Thermal Slowdown (reducing the core clocks by a factor of 2 or more) is engaged (temperature being too high) ('HW Thermal Slowdown' in nvidia-smi --query)",
+	reasonHWSlowdownThermal: {
+		description:  "HW Thermal Slowdown (reducing the core clocks by a factor of 2 or more) is engaged (temperature being too high) ('HW Thermal Slowdown' in nvidia-smi --query)",
+		isHWSlowdown: true,
+	},
 
 	// ref. nvmlClocksThrottleReasonHwPowerBrakeSlowdown
-	0x0000000000000080: "HW Power Brake Slowdown (reducing the core clocks by a factor of 2 or more) is engaged (External Power Brake Assertion being triggered) ('HW Power Brake Slowdown' in nvidia-smi --query)",
+	reasonHWSlowdownPowerBrake: {
+		description:  "HW Power Brake Slowdown (reducing the core clocks by a factor of 2 or more) is engaged (External Power Brake Assertion being triggered) ('HW Power Brake Slowdown' in nvidia-smi --query)",
+		isHWSlowdown: true,
+	},
 
 	// ref. nvmlClocksEventReasonDisplayClockSetting
-	0x0000000000000100: "GPU clocks are limited by current setting of Display clocks",
+	reasonDisplayClockSetting: {
+		description:  "GPU clocks are limited by current setting of Display clocks",
+		isHWSlowdown: false,
+	},
+}
+
+func (inst *instance) ClockEventsSupported() bool {
+	inst.mu.RLock()
+	defer inst.mu.RUnlock()
+
+	return inst.clockEventsSupported
+}
+
+func (inst *instance) RecvClockEventsHWSlowdown() <-chan *ClockEvents {
```

Comment:
seems currently no one is using this function

Reply:
right, removed